### PR TITLE
Fix deprecation warnings and make tests run again

### PR DESCRIPTION
**Pull request by LMescheder**
This pull requests fixes the current deprecation warnings and makes the tests run again. My editor also automatically removed quite a bit of extra white space, so the changes might seem bigger than they actually are...
After all tests have completed I still get the following error messages that I haven't figured out yet:

> error in running finalizer: OpenCL.CLError(code=-34, desc=:CL_INVALID_CONTEXT)
> error in running finalizer: OpenCL.CLError(code=-34, desc=:CL_INVALID_CONTEXT)

### CODE REVIEW

File: src/CLFFT.jl [hunk 1]
```diff
@@ -345,9 +345,9 @@ end
 
 dim(p::Plan) = begin
     res  = Int32[0]
-    size = Csize_t[0]
+    size = cl.CL_uint[0]
```
Comment (LMescheder) [inline src/CLFFT.jl:348]: Using `Csize_t` here resulted in an error on my system, as `Csize_t` translated to `UInt64` whereas `cl.CL_uint` translated to `UInt32`.

**Comment by dfdx** [comment]
Tests on Travis are failing, but they fail in latest master as well, so it should be unrelated to your PR. The rest looks fine for me. 

Could you please squash your changes to make git history a little bit more beautiful? Normally you can do like (from your branch): 

```
git fetch
git rebase -i master
```

and then mark with "pick" all commits you want to keep and with "s" all commits you want to squash into previous commit.

**Comment by LMescheder** [comment]
Ok, done.

**Comment by dfdx** [comment]
Thanks, merging it.

**Comment by tkelman** [comment]
You may want to switch to using `language: julia` rather than the Julia PPA so you can easily test different versions of Julia. On release Julia this might work better.

**Pull request merged**